client/src/app/shared/services/ontology.service.ts

```diff
@@ -49,18 +52,20 @@ export class OntologyService {
     return combineLatest(ontologyClasses
       .map(ontologyClass => ontologyClass.replace(':', '_'))
       .map(olsClass => this.select({q: olsClass}))
-    ).map(responses => responses
-      .map(ols => ols.response)
-      .filter(olsResponse => olsResponse.numFound === 1)
-      .map(olsResponse => olsResponse.docs[0].iri)
-    ).map(iriArray => {
-      searchParams[this.OLS_RELATION[ontologyRelation]] = iriArray;
-      return searchParams;
-    });
+    ).pipe(
+      map(responseArray => responseArray
+        .map(ols => ols.response)
+        .filter(olsResponse => olsResponse.numFound === 1)
+        .map(olsResponse => olsResponse.docs[0].iri)),
```

Comment:
Some operators here aren't piped e.g `ontologyClasses.map..` and `responseArray.map....`

Reply:
This occurs a few times (could do a Find-> .map for all occurrences)

Reply:
those are maps from arrays rather than Observables

Reply:
In which case is there a different convention to make that obvious?

Reply:
Ahh I see, that's fine. Could also do `of(responseArray).pipe(....)` to get back into observables land. Could also look at flatMap()-ing the nested map()s